python/simulation.py

```diff
@@ -859,6 +864,155 @@ def amplitude(self, point, component):
         return mp.eigenmode_amplitude(self.swigobj, swig_point, component)
 
 
+class Pade:
+    """
+    Pade is implemented as a class with a [`__call__`](#Pade.__call__) method,
+    which allows it to be used as a step function that collects field data from a given
+    point and runs [Pade](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.pade.html)
+    on that data to extract an analytic rational function which models the frequency response.
+
+    See [`__init__`](#Pade.__init__) for details about constructing a `Pade`.
+
+    In particular, Pade stores the discrete time series $\\hat{f}[n]$ corresponding to the given field
+    component as a function of time and expresses it as:
+
+    $$\\hat{f}(\\omega) = \\sum_n \\hat{f}[n] e^{i\\omega n \\Delta t}$$
+
+    The above is a "Taylor-like" polynomial in $$n$$ with a Fourier basis and
+    coefficients which are the sampled field data. We then compute the Pade approximant
+    to the analytic form of this function as:
+
+    $$R(\\omega) = \\frac{P(\\omega)}{Q(\\omega)}$$
+
+    Where $$P$$ and $$Q$$ are polynomials of degree $$m$$ and $$n$$, and $$m + n + 1$$ is the
+    degree of agreement of the Pade Approximant to the analytic function $$f(\\omega)$$. This
```

Comment:
typo: "Approximant" should be "approximant"

Reply:
https://github.com/NanoComp/meep/pull/2440/commits/6dd0939cac5f12cd11ece3b659ddd524f121f41e